### PR TITLE
Add whenever

### PR DESCRIPTION
**Pull request by vasya539**
- [x] merged with the latest master branch
- [x] resolved all the conflicts, PR could be merged
- [x] new code has new specs
- [ ] for new models - new factories were added too
- [x] unit specs are passed
- [x] code is well-formatted, Ruby, CSS, etc
- [x] screenshots of new pages, features are added
- [ ] code passed internal team code review
- [x] no instance variables in partials
- [ ] code is well formatted and has spaces!!!

PS: delete your branch in github after merge http://prntscr.com/q80mvq

### CODE REVIEW

File: app/mailers/super_admin_mailer.rb [hunk 1]
```diff
@@ -1,11 +1,20 @@
 class SuperAdminMailer < ApplicationMailer
   default from: 'remote-flow@mail.com'
+  @super_admins_email = User.super_admins.pluck(:email)
```
Comment (Sweex99) [inline app/mailers/super_admin_mailer.rb]: ВАСЯ чого!!!:? переніс це

File: app/mailers/super_admin_mailer.rb [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
-    admins_email = User.super_admins.pluck(:email)
-    return false if admins_email.empty?
-
     @organization = organization
-    mail(to: admins_email, subject: 'New organization created')
+    mail(to: @super_admins_email, subject: 'New organization created')
+  end
+
+  def send_statistic_about_new_users_and_organizations
+    @users_count = User.all(:conditions => ["? - created_at <= 7", Date.today]).count
```
Comment (Sweex99) [inline app/mailers/super_admin_mailer.rb]: перепиши на where

File: app/mailers/super_admin_mailer.rb [hunk 3]
```diff
@@ -1,11 +1,20 @@
 class SuperAdminMailer < ApplicationMailer
   default from: 'remote-flow@mail.com'
+  @super_admins_email = User.super_admins.pluck(:email)
+  before_each :check_presentce_of_super_admin
```
Comment (Sweex99) [inline app/mailers/super_admin_mailer.rb]: нахєра

File: app/mailers/super_admin_mailer.rb [hunk 4]
```diff
@@ -1,11 +1,20 @@
...
+    mail(to: @super_admins_email, subject: 'New organization created')
+  end
+
+  def send_statistic_about_new_users_and_organizations
+    @users_count = User.all(:conditions => ["? - created_at <= 7", Date.today]).count
+    @organizatinos_count = Organization.all(:conditions => ["? - created_at <= 7", Date.today]).count
+    mail(to: @super_admins_email, subject: 'Weekly statistics')
+  end
+
+  def check_presentce_of_super_admin
```
Comment (Sweex99) [inline app/mailers/super_admin_mailer.rb]: видаляй

File: app/views/super_admin_mailer/send_statistic_about_new_users_and_organizations.html.slim [hunk 5]
```diff
@@ -0,0 +1,10 @@
+doctype html
```
Comment (Sweex99) [inline app/views/super_admin_mailer/send_statistic_about_new_users_and_organizations.html.slim]: забери https://github.com/vikdotdev/remote-flow/blob/master/app/views/layouts/mailer.html.slim лишнє

File: app/mailers/super_admin_mailer.rb (hunk 1 above)
Comment (vasya539) [reply app/mailers/super_admin_mailer.rb]: це я не кидав ще на ревью навіть, я тоді тільки тестив як гем працює, в мене там все інше

File: app/mailers/super_admin_mailer.rb [hunk 6]
```diff
@@ -2,10 +2,19 @@ class SuperAdminMailer < ApplicationMailer
...
+
+  def send_statistic_about_new_users_and_organizations
+    @users_count = User.where(["? - created_at <= 7", Date.today]).count
+    @organizatinos_count = Organization.where(["? - created_at <= 7", Date.today]).count
+    mail(to: pluck_super_admin_emails, subject: 'Weekly statistics')
+  end
+
+  private
+
+  def pluck_super_admin_emails
```
Comment (demchyna) [inline app/mailers/super_admin_mailer.rb]: super_admin_emails

File: app/views/super_admin_mailer/send_statistic_about_new_users_and_organizations.html.slim [hunk 7]
```diff
@@ -0,0 +1,5 @@
+h1 Statistic about users:
+p
+  = @users_count + " users were created"
+p
+  = @organizations_count + " organiztions were created"
```
Comment (demchyna) [inline app/views/super_admin_mailer/send_statistic_about_new_users_and_organizations.html.slim]: pluralize

File: lib/tasks/send_statistic_to_super_admin.rake [hunk 8]
```diff
@@ -0,0 +1,6 @@
+namespace :send_statistic_to_super_admin do
+  desc 'Send super_admin statistic about new users and organizations'
+  task send_statistic_to_super_admin: :development do
+    SuperAdminMailer.send_statistic_about_new_users_and_organizations
```
Comment (demchyna) [inline lib/tasks/send_statistic_to_super_admin.rake]: test!

**Comment by vasya539** [comment]
image

File: spec/rake/rake_tasks_spec.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+require 'rake'
+Rails.application.load_tasks
+
```
Comment (igorkasyanchuk) [inline spec/rake/rake_tasks_spec.rb]: рейк таск потрібен?

File: app/mailers/super_admin_mailer.rb [hunk 10]
```diff
@@ -2,10 +2,19 @@ class SuperAdminMailer < ApplicationMailer
   default from: 'remote-flow@mail.com'
 
   def new_organization_email(organization)
-    admins_email = User.super_admins.pluck(:email)
-    return false if admins_email.empty?
-
     @organization = organization
-    mail(to: admins_email, subject: 'New organization created')
+    mail(to: super_admin_emails, subject: 'New organization created') unless super_admin_emails.empty?
```
Comment (romaslmd) [inline app/mailers/super_admin_mailer.rb]: method has small performance issue, multiple querying for super admin emails
prev version was better, except there's no need to explicitly return `false`, `nil` would work better

File: app/mailers/super_admin_mailer.rb [hunk 11]
```diff
@@ -2,10 +2,19 @@ class SuperAdminMailer < ApplicationMailer
...
+  end
+
+  def send_statistic_about_new_users_and_organizations
+    @users_count = User.where(["DATE(created_at) >= ?", Date.today - 7.days]).count
```
Comment (romaslmd) [inline app/mailers/super_admin_mailer.rb]: fyi: `7.days.ago.to_date`

File: app/mailers/super_admin_mailer.rb [hunk 12]
```diff
@@ -2,10 +2,19 @@ class SuperAdminMailer < ApplicationMailer
...
+    @organizatinos_count = Organization.where(["DATE(created_at) >= ?", Date.today - 7.days]).count
+    mail(to: super_admin_emails, subject: 'Weekly statistics') unless super_admin_emails.empty?
+  end
+
+  private
+
+  def super_admin_emails
+    User.super_admins.pluck(:email)
```
Comment (romaslmd) [inline app/mailers/super_admin_mailer.rb]: just in case, add `distinct`

File: app/mailers/super_admin_mailer.rb [hunk 13]
```diff
@@ -2,10 +2,19 @@ class SuperAdminMailer < ApplicationMailer
...
-
     @organization = organization
-    mail(to: admins_email, subject: 'New organization created')
+    mail(to: super_admin_emails, subject: 'New organization created') unless super_admin_emails.empty?
+  end
+
+  def send_statistic_about_new_users_and_organizations
+    @users_count = User.where(["DATE(created_at) >= ?", Date.today - 7.days]).count
+    @organizatinos_count = Organization.where(["DATE(created_at) >= ?", Date.today - 7.days]).count
+    mail(to: super_admin_emails, subject: 'Weekly statistics') unless super_admin_emails.empty?
```
Comment (romaslmd) [inline app/mailers/super_admin_mailer.rb]: note: when there're no super admin emails, then  there's no reason to query those counts above

File: spec/rake/rake_tasks_spec.rb [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+describe 'send_mail namespace rake tasks' do
+  describe 'send_mail:send_statistic_to_super_admin' do
+    Rake::Task.task_defined?("send_mail:send_statistic_to_super_admin")
```
Comment (romaslmd) [inline spec/rake/rake_tasks_spec.rb]: this thing really does nothing
1. no example definition
2. test fails only if error is raised within an example, not a context definition

**Pull request merged**